### PR TITLE
Remove hardcoded trigger bits from Lumi ALCARECOs

### PR DESCRIPTION
**Pull request by Moanwar**
#### PR description:

Remove hardcoded trigger bits from AlCaRecos and include it as a key in the alcareco trigger bits tag in "ALCARECOAlCaPCCZeroBias" and "ALCARECOAlCaPCCRandom"

Two GTs were created to be included in this PR, with the correct AlCaRecoTriggerBits tag.
1- 124X_dataRun3_Express_frozen_v6 :
GlobalTag for Run3 data relvals (express GT) - identical to 124X_dataRun3_Express_v6 but with snapshot at 2022-10-04 14:22:26 (UTC)
Here is the link of difference of GT, old vs new : https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/124X_dataRun3_Express_frozen_v4/124X_dataRun3_Express_frozen_v6

2- 124X_dataRun3_Prompt_frozen_v5 :
GlobalTag for Run3 data relvals (prompt GT) - identical to 124X_dataRun3_Prompt_v5 but with snapshot at 2022-10-04 14:19:51 (UTC)
Here is the link of difference of GT, old vs new : https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/124X_dataRun3_Prompt_frozen_v4/124X_dataRun3_Prompt_frozen_v5

#### PR validation:
Not a backport, but will be backported for data-taking operations

@cms-AlCaDB/alca-tsg

### CODE REVIEW

**Comment by Moanwar** [comment]
adding  @tvami , @francescobrivio

File: Calibration/LumiAlCaRecoProducers/python/ALCARECOAlCaPCCRandom_cff.py [hunk 1]
```diff
@@ -2,8 +2,8 @@
 
 import HLTrigger.HLTfilters.hltHighLevel_cfi
 ALCARECORandomHLT = HLTrigger.HLTfilters.hltHighLevel_cfi.hltHighLevel.clone(
-    HLTPaths = cms.vstring("AlCa_LumiPixelsCounts_Random_v*"),
-    eventSetupPathsKey='',
+    #HLTPaths = cms.vstring("AlCa_LumiPixelsCounts_Random_v*"),
+    eventSetupPathsKey='AlCa_LumiPixelsCounts_Random_v*',
```
Comment (tvami) [inline Calibration/LumiAlCaRecoProducers/python/ALCARECOAlCaPCCRandom_cff.py:6]: I dont think this is the name of the key, is it?

File: Calibration/LumiAlCaRecoProducers/python/ALCARECOAlCaPCCZeroBias_cff.py [hunk 2]
```diff
@@ -2,8 +2,8 @@
 
 import HLTrigger.HLTfilters.hltHighLevel_cfi
 ALCARECOZeroBiasHLT = HLTrigger.HLTfilters.hltHighLevel_cfi.hltHighLevel.clone(
-    HLTPaths = cms.vstring("AlCa_LumiPixelsCounts_ZeroBias_v*"),
-    eventSetupPathsKey='',
+    #HLTPaths = cms.vstring("AlCa_LumiPixelsCounts_ZeroBias_v*"),
+    eventSetupPathsKey='AlCa_LumiPixelsCounts_ZeroBias_v*',
```
Comment (tvami) [inline Calibration/LumiAlCaRecoProducers/python/ALCARECOAlCaPCCZeroBias_cff.py:6]:
```suggestion
    eventSetupPathsKey='AlCaPCCZeroBias',
```

File: Calibration/LumiAlCaRecoProducers/python/ALCARECOAlCaPCCRandom_cff.py (hunk 1 above)
Comment (tvami) [reply Calibration/LumiAlCaRecoProducers/python/ALCARECOAlCaPCCRandom_cff.py:6]:
```suggestion
    eventSetupPathsKey='AlCaPCCRandom',
```
I think this is what we need and then a new trigger bit in the GT

**Comment by ChrisMisan** [comment]
@Moanwar kind reminder

**Comment by ChrisMisan** [comment]
@Moanwar ping

**Comment by malbouis** [comment]
test parameters:

- workflows = 1020,1002.3,1002.4,1001.3

**Comment by malbouis** [comment]
please test

**Comment by tvami** [comment]
I think we need to retest, this shows differences that cannot come from this PR, e.g. https://tinyurl.com/2odjehov

**Comment by tvami** [comment]
@Moanwar please update the PR: 
 * the title should reflect that this is for Lumi ALCARECOs only
 * the description should include the change of GT as well, and also a link to the diff of GT, old vs new
 
 thanks

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
> and also a link to the diff of GT, old vs new

like this for the Prompt:
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/124X_dataRun3_Prompt_frozen_v4/124X_dataRun3_Prompt_frozen_v5

and then looking at this... maybe the IOV changes in the ECAL conditions are responsible for https://github.com/cms-sw/cmssw/pull/39519#issuecomment-1270460534

**Comment by malbouis** [comment]
please test

**Comment by tvami** [comment]
+alca
 * tests pass, diffs are due to the change of snapshot time

File: Calibration/LumiAlCaRecoProducers/python/ALCARECOAlCaPCCRandom_cff.py [hunk 3]
```diff
@@ -2,8 +2,8 @@
 
 import HLTrigger.HLTfilters.hltHighLevel_cfi
 ALCARECORandomHLT = HLTrigger.HLTfilters.hltHighLevel_cfi.hltHighLevel.clone(
-    HLTPaths = cms.vstring("AlCa_LumiPixelsCounts_Random_v*"),
-    eventSetupPathsKey='',
+    #HLTPaths = cms.vstring("AlCa_LumiPixelsCounts_Random_v*"),
+    eventSetupPathsKey='AlCaPCCRandom',
     TriggerResultsTag = cms.InputTag("TriggerResults","","HLT"),
     andOr = cms.bool(True), # choose logical OR between Triggerbits
     throw = cms.bool(False) # tolerate triggers stated above, but not available
```
Comment (perrotta) [inline Calibration/LumiAlCaRecoProducers/python/ALCARECOAlCaPCCRandom_cff.py:9]:
```suggestion
    #HLTPaths = ["AlCa_LumiPixelsCounts_Random_v*"],
    eventSetupPathsKey='AlCaPCCRandom',
    TriggerResultsTag = ("TriggerResults","","HLT"),
    andOr = True, # choose logical OR between Triggerbits
    throw = False # tolerate triggers stated above, but not available
```

File: Calibration/LumiAlCaRecoProducers/python/ALCARECOAlCaPCCZeroBias_cff.py [hunk 4]
```diff
@@ -2,8 +2,8 @@
 
 import HLTrigger.HLTfilters.hltHighLevel_cfi
 ALCARECOZeroBiasHLT = HLTrigger.HLTfilters.hltHighLevel_cfi.hltHighLevel.clone(
-    HLTPaths = cms.vstring("AlCa_LumiPixelsCounts_ZeroBias_v*"),
-    eventSetupPathsKey='',
+    #HLTPaths = cms.vstring("AlCa_LumiPixelsCounts_ZeroBias_v*"),
```
Comment (perrotta) [inline Calibration/LumiAlCaRecoProducers/python/ALCARECOAlCaPCCZeroBias_cff.py:5]: same here

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
+1
 * only change is in the types

**Comment by perrotta** [comment]
+1

**Comment by perrotta** [comment]
@malbouis thank you for telling
Chenges in that file are coming from PR #39471, which was merged right after this PR was first submitted. It doesn't look like this PR has modified anything that was coded in #39471.
I don't know how did it happen, I can only suspect that having based this PR to the wrong branch [at the beginning](https://github.com/cms-sw/cmssw/pull/39519#issuecomment-1260472352) could have brought something extra with it once automatically rebased to the master.
In any case, I don't see issue with that file or whatever else got touched by this PR.

**Pull request merged**